habitat/robots/static_manipulator.py

```diff
@@ -65,424 +65,51 @@ def __init__(
-        self.arm_joint_pos = fix_arm_joint_pos
-
-    @property
-    def arm_velocity(self) -> np.ndarray:
-        """Get the velocity of the arm joints."""
-
-        # deref self vars to cut access in half
-        joint_dof_indices = self.joint_dof_indices
-        arm_joints = self.params.arm_joints
-        sim_obj_joint_vel = self.sim_obj.joint_velocities
-
-        arm_dof_indices = (joint_dof_indices[x] for x in arm_joints)
-        return np.array(
-            [sim_obj_joint_vel[i] for i in arm_dof_indices],
-            dtype=np.float32,
-        )
-
-    @property
-    def arm_motor_pos(self) -> np.ndarray:
-        """Get the current target of the arm joints motors."""
-        motor_targets = np.zeros(len(self.params.arm_init_params))
-        for i, jidx in enumerate(self.params.arm_joints):
-            motor_targets[i] = self._get_motor_pos(jidx)
-        return motor_targets
-
-    @arm_motor_pos.setter
-    def arm_motor_pos(self, ctrl: List[float]) -> None:
-        """Set the desired target of the arm joint motors."""
-        self._validate_arm_ctrl_input(ctrl)
-
-        for i, jidx in enumerate(self.params.arm_joints):
-            self._set_motor_pos(jidx, ctrl[i])
-
-    def clip_ee_to_workspace(self, pos: np.ndarray) -> np.ndarray:
-        """Clips a 3D end-effector position within region the robot can reach."""
-        return np.clip(
-            pos,
-            self.params.ee_constraint[:, 0],
-            self.params.ee_constraint[:, 1],
-        )
+        Manipulator.update(self)
+        RobotBase.update(self)
 
-    @property
-    def arm_motor_forces(self) -> np.ndarray:
-        """Get the current torques on the arm joint motors"""
-        return np.array(self.sim_obj.joint_forces)
-
-    @arm_motor_forces.setter
-    def arm_motor_forces(self, ctrl: List[float]) -> None:
-        """Set the desired torques of the arm joint motors"""
-        self.sim_obj.joint_forces = ctrl
-
-    #############################################
-    # BASE RELATED
-    #############################################
-
-    @property
-    def base_pos(self):
-        """Creates base position property, should not change"""
-        return mn.Vector3()  # zeros
-
-    @base_pos.setter
-    def base_pos(self, position: mn.Vector3):
-        """Attempts to change the base position will result in an error to prevent accidental mobility"""
-        raise NotImplementedError(
-            "Setting the base position of a static manipulator is not permitted."
-        )
-
-    @property
-    def base_rot(self) -> float:
-        return self.sim_obj.rotation.angle()
-
-    @base_rot.setter
-    def base_rot(self, rotation_y_rad: float):
-        raise NotImplementedError(
-            "Setting the base rotation of a static manipulator is not permitted."
-        )
-
-    @property
-    def base_transformation(self):
-        return self.sim_obj.transformation
-
-    #############################################
```

Comment:
Ow, it was already used before, anyway let's fix it.

Reply:
Done, thank you!